3_javascript/javascript3/week3/screenshot_app/script.js

```diff
@@ -0,0 +1,151 @@
+domListener();
+function domListener() {
+  document.addEventListener("DOMContentLoaded", domLoader);
+}
+
+function domLoader() {
+  const inputURL = document.getElementById("inputURL");
+  const takeURLButton = document.getElementById("takeURLButton");
+  const showScreenshot = document.getElementById("showURLButton");
+  const selectScreenshot = document.getElementById("selectScreenshot");
+  const deleteSelectedScreenshotButton = document.getElementById(
+    "deleteSelectedScreenshotButton"
+  );
+
+  inputURL.addEventListener("input", () => (URL = inputURL.value));
+  takeURLButton.addEventListener("click", () => screenshotGenerator(URL));
+  showScreenshot.addEventListener("click", loadScreenshots);
+  deleteSelectedScreenshotButton.addEventListener(
+    "click",
+    deleteSelectedScreenshot
+  );
+
+  function screenshotGenerator(URL) {
+    return new Promise((resolve) => {
+      const loadingIndicator = document.getElementById("loadingIndicator");
+      loadingIndicator.style.display = "block";
+      const screenshotURL = getScreenshotURL(URL);
+      const urlAPI = `https://website-screenshot6.p.rapidapi.com/screenshot?url=https%3A%2F%2F${screenshotURL}&width=1920&height=1080&fullscreen=true`;
+      const options = {
+        method: "GET",
+        headers: {
+          "X-RapidAPI-Key":
+            "295c037658msh02afd26749f76c3p1c8a77jsne1bb5ae68f4d",
+          "X-RapidAPI-Host": "website-screenshot6.p.rapidapi.com",
+        },
+      };
+      fetch(urlAPI, options)
+        .then((data) => data.json())
+        .then((screenshot) => {
+          if (screenshot.screenshotUrl) {
+            saveScreenshot(screenshot);
+          } else {
+            alert("The URL is not valid.");
+          }
+        })
+        .catch((error) => console.error("Error", error))
+        .finally(() => {
+            loadingIndicator.style.display = "none";
+          });
+    });
+  }
+
+  function getScreenshotURL(URL) {
+    return URL;
```

Comment:
It seems like this function doesn't do anything, meaning you can just delete the function and use the `URL` variable by itself.

Reply:
So at line 27 it would be:
`const screenshotURL = URL;`